synapse/logging/opentracing.py

```diff
@@ -909,8 +909,9 @@ def tag_args(func: Callable[P, R]) -> Callable[P, R]:
     @wraps(func)
     def _tag_args_inner(*args: P.args, **kwargs: P.kwargs) -> R:
         argspec = inspect.getfullargspec(func)
-        for i, arg in enumerate(argspec.args[1:]):
-            set_tag("ARG_" + arg, str(args[i]))  # type: ignore[index]
+        # We use `[1:]` to skip the `self` object reference
+        for i, arg in enumerate(args[1:]):
```

Comment:
Patrick and I saw something similar here: https://github.com/matrix-org/synapse/pull/13345/files#r925922374

You might prefer to use `enumerate(args[1:], start=1)` so that you can write `argspec.args[i]`below. Six and two threes though.

Reply:
Oh, this is actually the same invocation! (I thought this was a different function :facepalm:)